docs/develop/architecture/event_handling.md

```diff
@@ -0,0 +1,69 @@
+# Event handling
+
+InvenioRDM backend is not event driven. At the time of writing (v11) there is no
+support for event handling, although there is work being done to implement an
+_event bus_ ([RFC](https://github.com/inveniosoftware/rfcs/pull/56)).
+
+Nonetheless, there are several addressed use cases that would benefit from an event bus
+and will be refactored to use it once it is implemented.
+
+In the following sections these specific use cases will be described with more detail.
+However, they all share a similar architecture.
+
+## Emitter
+
+The object in charge of sending/triggering the "event". The lack of an event bus leaves
+freedom of implementation on the emitter. For example, they could be emitted from the
+views, after unit of work (record) commit, etc. These events are written to case-specific
+message queues.
+
+The important factor to have into account is the decoupling of the _happening_ of the
+action and creating the _event about it happening_. For example, notifying about record
+creation should be done after the record has been committed. This would avoid
+inconsistencies such as failing to commit a record because the event failed to be created,
+or sending a notification about something that was not committed yet and might fail to do
+so. This leads to the question of _what happens if after commit I cannot create the event?_
+and other corner cases. Addressing these cases is out of the scope of the current
+implementations and will be solved by the event bus.
+
+## Handler
+
+For every event there is one or more operations that need to be executed. In the following
+examples the logic is implemented in case-specific tasks that run periodically (celery
+beat).
+
+## Events
+
+Since there is no defined event API, the current use cases define case-specific payloads
+and formats.
+
+## Examples
+
+### Change notifications
+
+The full design details can be found in the [RFC](https://github.com/inveniosoftware/rfcs/blob/master/rfcs/framework-0062-change-notification.md).
+
+InvenioRDM uses polyglot persistence: PostgreSQL and OpenSearch. Due to data retrieval and
+search requirements data is denormalized in OpenSearch. This means that when some
+entities  are updated (e.g. user profiles) objects containing their denormalized
+information also need to be updated too (e.g. membership requests reflecting the user's
+username).
+
+To achieve this, the events or _change notifications_ are registered in the unit of work.
+On an operation that will trigger an asynchronous celery task with a custom payload on
+`post_commit`. This task will enrich the payload and trigger all the handlers registered
+for the record type that was changed. These handlers are service methods with custom
+signatures, and are configurable in the `invenio.cfg` file.
+
+### Statistics
+
+Statistics such as record view or file downloaded are calculated using events. These
```

Comment:
```suggestion
Statistics such as record view or file downloads are calculated using events. These
```

Not sure if you meant number of file downloads or not, leaving the suggestion nonetheless